search.php

```diff
@@ -38,7 +38,51 @@
 ?>
 		<article class="search-card pb">
 			<a href="<?= esc_url( get_permalink() ); ?>"><?= esc_html( get_the_title() ); ?></a>
-			<div><?= get_the_excerpt(); ?></div>
+			
+			
+			   <?php 
+				// if the post/page has no excerpt, get_the_excerpt returns an empty string and thus an empty div
+				// so check that the exceprt exists	   
+				if(get_the_excerpt() ) { 
+				
+				?>
+			   
+				<div><?= the_excerpt(); ?></div>
+				<?php 
+			    } else {  
+			
+				// if there's no excerpt, check the ACF fields for the first "content" block and give me the first 250 words with an ellipses at the end
+				if ( have_rows( 'main_blocks' ) ){

```

Comment:
Would it be feasible to extract this to a separate function rather than having it inlined in the search template?

Reply:
Probably? I suppose I could extract this and put it in `functions.php` then invoke it here. Maybe I should try to emulate what Phire has already done, which is use template pieces and include them in the template. My way is a bit messier, but the Phire way means you have to know where things are in order to change/fix them. It's not really a problem either way, just different ways of doing things.

Reply:
WordPress recently changed how templates work, but Phire (and most of my WP knowledge) uses standards from before those changes. You can see how the official theme from 2021 has the `while` "Loop" in the template files:

https://github.com/WordPress/twentytwentyone/blob/trunk/single.php
https://github.com/WordPress/twentytwentyone/blob/trunk/search.php

It's just how things were done at the time. We could probably work towards updating our templates to reflect the new standards if that's something we want to focus on for the future.